src/NSwag.CodeGeneration.CSharp/Templates/ClientTemplate.tt

```diff
@@ -113,10 +113,20 @@ public partial interface I<#=Model.Class#>
     }
 
 <#}#>
+    partial void UpdateJsonSettings(ref Newtonsoft.Json.JsonSerializerSettings settings);
```

Comment:
Don't use ref as it is already a reference and with ref you could completely replace the settings...